### PR TITLE
Docs How to Improve performance of flyte

### PR DESCRIPTION
**Pull request by kumare3**
Signed-off-by: Ketan Umare <ketan.umare@gmail.com>

### CODE REVIEW

File: rsts/howto/performance/index.rst [hunk 1]
```diff
@@ -4,6 +4,90 @@
...
+.. note:: As you increase the number of workers in FlytePropeller it is important to increase the number of cpu's given to flytepropeller pod.
+
+
+Another area of slowdown could be the size of the input-output cache that Flytepropeller maintains in-memory. This can be configured, while configuring
+the storage for FlytePropeller.
+
+Rule of thumb, for Flytepropeller that has 2GB or memory, should allocate about 512MB-756MB to the input/output cache.
+
+
+Scaling out FlyteAdmin
```
Comment (EngHabu) [inline rsts/howto/performance/index.rst:75]: When should they scale vertically vs horizontally? maybe link to horizontal autoscaler? is there a good metric to use to scale up horizontally? Maybe request rate?

File: rsts/howto/performance/index.rst [hunk 2]
```diff
@@ -4,6 +4,90 @@
...
+=======================
+Flyteadmin is a stateless service. Often time before needing to scale FlyteAdmin, you need to scale the backing database. Check out the FlyteAdmin Dashboard to see signs of latency degredation and increase the size of backing postgres instance.
+FlyteAdmin is a stateless service and its replicas (in the kubernetes deployment) can be simply increased to allow higher throughput.
+
+Scaling out Datacatalog
```
Comment (EngHabu) [inline rsts/howto/performance/index.rst:80]: ditto

File: rsts/howto/performance/index.rst [hunk 3]
```diff
@@ -4,6 +4,90 @@
...
+==========================
+`FlytePropeller <https://pkg.go.dev/github.com/flyteorg/flytepropeller>`_ is the core engine of Flyte that executes the workflows for Flyte. It is implemented as a `controller <https://kubernetes.io/docs/concepts/architecture/controller/>`_ in Kubernetes.
+FlytePropeller has a lot of knobs that can be tweaked for performance. The default configuration is good enough for small to medium sized installations of Flyte, that are running about 500 workflows concurrently with no noticeable overhead. In the case when the number of workflows increases,
+FlytePropeller will automatically slow down, without losing correctness.
+
+Here are signs of slowdown,
+
+#. Round Latency for each workflow increases
+#. Transition latency increases
+#. Workflows take longer to start
```
Comment (EngHabu) [inline rsts/howto/performance/index.rst:19]: Links to these metrics in grafana dashboard?

File: rsts/howto/performance/index.rst [hunk 4]
```diff
@@ -4,6 +4,90 @@
...
+
+This is usually because the number of threads in FlytePropeller are not enough to keep up with the number of workflows. This can be resolved by adjusting FlytePropeller config specified `here <https://pkg.go.dev/github.com/flyteorg/flytepropeller@v0.10.3/pkg/controller/config>`_.
+
+.. list-table:: Important Properties
```
Comment (EngHabu) [inline rsts/howto/performance/index.rst:23]: I think for every (most of) property here, we should link to a chart that shows the effect?

Comment (kumare3) [reply rsts/howto/performance/index.rst:23]: how to link to a chart?

Comment (EngHabu) [reply rsts/howto/performance/index.rst:23]: maybe like [this](https://github.com/flyteorg/flyte/blob/master/stats/flytepropeller_dashboard.py#L51-L63)

**Pull request closed without merging**